### DISCUSSION TITLE
ReactPy - The Rebrand is Real!

### DISCUSSION THREAD

**Discussion by rmorshea** [idom-team/idom#882] [Announcements]
In the next month or so this project will be renamed to ReactPy 🎉 

The change will happen once we've completed all tasks from our [1.0 milestone](https://github.com/idom-team/idom/milestone/1). Afterwards there will be a final `1.0` release of IDOM along with any necessary follow-up patch releases. From there we will begin the process of renaming the project to ReactPy and moving it to a new repo [on GitHub](https://github.com/reactive-python/reactpy) and [on PyPI](https://pypi.org/project/ReactPy/). The initial release of ReactPy will be identical to IDOM's 1.0 release. So to start out, you'll be able to swap any usages `idom` for `reactpy` without any trouble.

Let us know if you have any questions or comments about this change.

**Comment by Archmonger** [comment]
We probably should do this via a repo transfer + rename to retain github stars and commit history.

**Reply by rmorshea** [reply]
Yeah, I mostly created the repo so the link would go somewhere, but in retrospect that might be confusing.

**Comment by rmorshea** [comment]
Also acquired a `reactpy` organization on npmjs. This should allow us to create scoped packages under this name (e.g. `@reactpy/client`).

**Comment by rmorshea** [comment]
We're going to need a new logo! I have some ideas:

<div>
<img src=https://user-images.githubusercontent.com/4041990/220450566-0e07edf3-b6dc-440c-9fe4-68543eb539f8.jpg width="48%"/>
<img src=https://user-images.githubusercontent.com/4041990/220450577-791948c3-0a03-4fef-967f-3e8a49e34571.jpg width="48%"/>
</div>

_The one on the left being the react logo, but with snakes instead of electron paths. The right is using the erlenmeyer flask to imply a "reaction" which connects with the word "react". I think the one on the right is easier to read, but the one of the left gets more at what the project is doing. The lack of readability of the left could just be due to my lacking artistic abilities though._

But I don't have the skills to make these look more "professional". I know that @acivitillo had said that his team had a designer we might be able to use, but I'm not actually sure if they're still using this project (which if you see this Alessio, you're sponsoring the project, feel free to pull that if you aren't anymore).

Anyone have suggestions on what we should do?

**Reply by rmorshea** [reply]
This might actually fine:

![reactpy](https://user-images.githubusercontent.com/4041990/220508331-02256bcb-a17d-4e30-973d-e5b4d812e794.png)

**Reply by Archmonger** [reply]
I'll take a quick crack at making a logo after PR review

**Reply by Archmonger** [reply]
Okay so stable diffusion and midjourney image generation sucked ass.  I'll go old-school photoshop and see what I can create.

Midjourney: 
![image](https://user-images.githubusercontent.com/16909269/220513979-c368c799-5183-4ca6-8780-e09bcb4ed448.png)

Stable Diffusion:
![image](https://user-images.githubusercontent.com/16909269/220514042-607cce36-7f30-4dea-a788-b6e6fd1e2473.png)
![image](https://user-images.githubusercontent.com/16909269/220514149-78c4a6a9-6452-4444-b557-651f56dacc65.png)

**Reply by rmorshea** [reply]
Yeah, I gave those a shot. They really don't seem to understand abstract logos.

**Reply by Archmonger** [reply]
I created a draft logo using your first concept as a basis. At minimum, I think the snake neck needs some touchup in the next revision.

![ReactPy Logo v2](https://user-images.githubusercontent.com/16909269/220589689-09418c6c-0ca8-4160-bfd9-d3aa8e7ef3e6.png)

EDIT: Here's a short tailed version
![ReactPy Logo v3](https://user-images.githubusercontent.com/16909269/220596143-b95ba269-3f2d-4d71-9e64-5e5f5c633741.png)

**Reply by rmorshea** [reply]
I really like this one.
<img src="https://user-images.githubusercontent.com/4041990/220789619-608fe9ff-56d9-4e5f-b87d-48c9c59fdf79.svg" width="75%"/>

**Reply by Archmonger** [reply]
LGMT. Should we consider matching the colors to the original react logo?

**Reply by rmorshea** [reply]
I think we can go for our own theme. Also don't want to invite any sort of legal questions.

**Reply by Archmonger** [reply]
Alright. For differentiation, how about green for Python (animal)?

Or perhaps one of the colors out of the Python logo

**Reply by rmorshea** [reply]
Subtle improvements

<img src="https://user-images.githubusercontent.com/4041990/220799726-c6f60b3a-dc31-448a-82fd-ba94b5bb5e63.svg" width="60%"/>

**Reply by rmorshea** [reply]
The Python logo has a bit of a gradient to it, bit took this blue from part of it. I'd be good with some sort of green though.

**Reply by rmorshea** [reply]
<img src="https://user-images.githubusercontent.com/4041990/220801461-fd17396f-788c-4f84-8c0d-571c6e98a55c.svg" width="60%" />

**Reply by rmorshea** [reply]
I think green looks good and is a nice way to differentiate since the logos look very similar at small scales.

**Reply by rmorshea** [reply]
more of a green/purple(ish) theme:

<img src="https://user-images.githubusercontent.com/4041990/220802863-c2ac30af-01d2-4ba7-8008-15212f6aa7d0.svg" width="60%"/>

**Reply by Archmonger** [reply]
Background color isn't too important, since we're almost always going to use it with transparency.

The green looks solid.

**Reply by numpde** [reply]
These are cool ideas. If you allow for a few days, I'll commission some folks on Fiverr to come up with concepts and critique. Furthermore, I'd be happy to produce an short explainer/animation (via https://randompile.net/). 

My own sense is that the last proposal is cool but doesn't quite pop.

**Reply by rmorshea** [reply]
That would be great! The linked images are SVGs (with some Inkscape specific xml) so you can send those over to whoever you commission. I quite like the latest design, but it could probably use some improvements to make it more readable when scaled down. Someone also suggested finding a way to animate the snakes so the my orbit around the center of the atom, say on-hover, which would be super cool. I don't have the CSS chops to do that though.

As for the explainer, that would be great to do in the future, but our efforts might be better spent in the near term finishing the documentation (happy to help explain the roadmap there if you want to contribute). If you'd rather work on the explainer though, I'd be able to provide feedback, but probably won't have time to write the script myself. At least in the next couple months.

**Reply by numpde** [reply]
> As for the explainer, that would be great to do in the future

Ok, let's stash that.

Would you mind defining a "brief" for the logo? to the tune of: an overview of the brand, description of the project, goals, target audience, competitors, style preferences, deliverables, budget, project timing, examples of existing design work, and what to avoid [[ref](https://archive.ph/QZF8D)]. 

Incidentally, do you know what was the rationale for the react logo itself? Besides "[thinking atomic](https://youtu.be/8pDqJVdNa44&t=56m14s)" (at 56m 14s).

**Reply by rmorshea** [reply]
I think the name "React" and the logo are a reference to chemical reactions. At least, that's how I read it.

**Reply by rmorshea** [reply]
The https://www.electronjs.org/ logo might be another source of inspiration.

**Reply by Archmonger** [reply]
I would imagine the largest thing is the logo should summarize the "merged" relationship between ReactJS and Python.

**Reply by rmorshea** [reply]
What do people think about going forward with the existing logo, but holding off on "launching" (i.e. making a release and announcing it)? At the moment, we're kind of in a frozen state where we can't make contributions (at least to core) until the project has been renamed.

**Reply by Archmonger** [reply]
Any logo is fine to move forward with, we can change it again after the rename.

**Reply by rmorshea** [reply]
@numpde, for now we're going to move forward with this draft of the logo, but we can make changes before we actually release.

**Reply by numpde** [reply]
> [..], I'll commission some folks on Fiverr to come up with concepts and critique. [..]

.. didn't get good enough ideas for the budget.

**Reply by Archmonger** [reply]
### An overview of the brand
ReactPy is a library for building user interfaces in Python without Javascript. ReactPy interfaces are made from [components](https://reactpy.dev/docs/guides/creating-interfaces/your-first-components/index.html#your-first-components) which look and behave similarly to those found in [ReactJS](https://reactjs.org/). Designed with simplicity in mind, ReactPy can be used by those without web development experience while also being powerful enough to grow with your ambitions.

### Description of the project
ReactJS for Python developers. This effectively allows users to develop websites while minimizing JavaScript code, but retaining the ease of development of ReactJS.

### Goals

- Converting all ReactJS features into Python equivalents
- Bridge the gap between JavaScript and Python web development
- Simplify using native ReactJS code within ReactPy
- Ability to perform all common browser-related features directly within Python.

### Target audience
Python developers who dislike, or don't know, JavaScript.

Amongst those who don't know Javascript, a fair portion are likely interested in building data applications or dashboards.

### Competitors

Generic frameworks:

- [Reacton](https://github.com/widgetti/reacton)
- [HTMX](https://htmx.org/)
- [Django Unicorn](https://www.django-unicorn.com/)
- [Django Sockpuppet](https://github.com/jonathan-s/django-sockpuppet)
- [Flask Meld](https://www.flask-meld.dev/)

Data application/dashboard focused frameworks:

- [Streamlit](https://streamlit.io/)
- [Panel](https://panel.holoviz.org/)
- [Plotly Dash](https://dash.plotly.com/introduction)

Partially source from: https://www.django-unicorn.com/docs/#comparison

### Style preferences
No strict style guidelines. But ideally, this icon should be uniquely recognizable within a favicon format.
We are currently leaning towards a green color scheme because cartoon snakes/pythons are green. 

We are open to color scheme and design suggestions. All existing mockups exist only as the first concept design we thought up.

### Deliverables

- Vectorized logo
- Vectorized logo with project name
- Vectorized favicon variant

### Budget
No strict budget, but this project is OSS/unfunded so keeping the costs low is ideal.

### Project timing
No strict timetables due to this being OSS/unfunded

### Examples of existing design work
[Ryan's snake variant](https://raw.githubusercontent.com/reactive-python/reactpy/2fc8228d176c42fc402202b6c6b7e6431d7f58c5/branding/svg/reactpy-logo-square.svg) and [Mark's simplified snake variant](https://user-images.githubusercontent.com/16909269/220596143-b95ba269-3f2d-4d71-9e64-5e5f5c633741.png).

### What to avoid
Avoid logos that are too closely identical in both color and design to ReactJS or Python. Basically, if a browser has two tabs open we want at least some visual indication that our tab is unique.

**Reply by rmorshea** [reply]
I added some of my thoughts. Overall this seems like a fair summary.

**Comment by rmorshea** [comment]
Ok, still work to be done to rename and move the other `idom-*` projects. But `idom` has finally been renamed to `reactpy`! With that said, we're going to hold off on an initial release until everything else is ready to go (new logo, other packages, etc).

**Reply by Archmonger** [reply]
On that note, looks like both the old docs link and the new one completely broke during the migration.

**Reply by rmorshea** [reply]
Docs (at least till I link them up to reactpy.dev) can be found here:

http://reactpy.herokuapp.com

**Reply by Archmonger** [reply]
The link in the readme is broken then.
